### DISCUSSION TITLE
Embedding - is there any solution for single URL allowlist twitter style iframe emebbing?

### DISCUSSION THREAD

**Discussion by JamesBWard** [mastodon/mastodon#22564] [General]
Hi, today many websites embed tweets by detecting a twitter URL and creating an iframe that displays that tweet. The equivalent embedding strategy is difficult with Mastodon because each instance has a different URL. Is there any existing overlay service that would allow a single URL to be used for this?

For example, imagine a URL like "https://mastodon.social/embed/tech.lgbt/@exampleUser/0123456789/embed". It could be any domain, for example https://my.new.aggregator.service.com/embed/mastodon.social/@exampleUser/0123456789.

Such a service would have potential downsides, for example it works against a decentralization goal and it could be expensive to run if it were widely adopted depending on the implementation and if the new service owner had to host content like videos and images. However, I can't spot any obvious issues that would prevent it from working. 

I would love to hear people's thoughts about this! Let me know what you think.

**Comment by nightpool** [comment]
No, we recommend either using the embed URL generated by the instance, or by requesting the URL and using the OpenGraph tags, the OEmbed endpoint, or the ActivityPub representation (which is included as a link head tag with rel=alternate)

Can you explain a little bit more about your usecase for this? I might be able to give some more detailed advice.

**Reply by JamesBWard** [reply]
Thanks for the reply! I post on a few forums that use xenforo and vbulletin, and when a user makes a post that links to twitter, youtube, instagram, etc, that link is automatically expanded into an embed. This lets users post content and other users can discuss the content locally or choose to click through and engage on those sites. 

The per-instance embed feature already works perfectly for this but the problem becomes allowing the parser to understand what it should expand and what is just a flat link. It's a content moderation risk to try to expand any link generically or to let users directly post html to embed, so that is where a multi-server allowlist comes up. However the admins of forums where I've floated this have been reluctant to maintain a quickly mutating(?) list of 3rd party servers.
